internal/kubernetes/operator/install_resources.go

```diff
@@ -0,0 +1,364 @@
+				namespaces = append(namespaces, watch...)
+			}
+		}
+	}
+
+	for _, watchNamespace := range namespaces {
+		obj := &rbacv1.Role{}
+		err := ir.objConverter.FromUnstructured(config, obj)
+		if err != nil {
+			return fmt.Errorf("failed to convert Role object: %w", err)
+		}
+
+		obj.SetNamespace(watchNamespace)
+
+		err = ir.kubeCtl.Create(ctx, obj)
+		if err != nil {
+			return fmt.Errorf("failed to add Role into cluster: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (ir *InstallResources) addClusterRole(ctx context.Context, config map[string]interface{}, namespace string) error {
+	obj := &rbacv1.ClusterRole{}
+	err := ir.objConverter.FromUnstructured(config, obj)
+	if err != nil {
+		return fmt.Errorf("failed to convert ClusterRole object: %w", err)
+	}
+
+	obj.SetNamespace(namespace)
+
+	err = ir.kubeCtl.Create(ctx, obj)
+	if err != nil {
+		return fmt.Errorf("failed to add CluserRole into cluster: %w", err)
+	}
+
+	return nil
+}
+
+func (ir *InstallResources) addRoleBindings(ctx context.Context, config map[string]interface{}, namespace string, watch []string) error {
+	namespaces := []string{namespace}
+
+	if value, ok := config["metadata"]; ok {
+		if metadata, ok := value.(map[string]interface{}); ok {
+			if name, ok := metadata["name"]; ok && name != leaderElectionRoleBindingName {
+				namespaces = append(namespaces, watch...)
+			}
+		}
+	}
+
+	for _, watchNamespace := range namespaces {
+		obj := &rbacv1.RoleBinding{}
+		err := ir.objConverter.FromUnstructured(config, obj)
+		if err != nil {
+			return fmt.Errorf("failed to convert RoleBinding object: %w", err)
+		}
+
+		obj.SetNamespace(watchNamespace)
+		obj.Subjects[0].Namespace = namespace
+
+		err = ir.kubeCtl.Create(ctx, obj)
+		if err != nil {
+			return fmt.Errorf("failed to add RoleBindig into cluster: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (ir *InstallResources) addClusterRoleBinding(ctx context.Context, config map[string]interface{}, namespace string) error {
+	obj := &rbacv1.ClusterRoleBinding{}
+	err := ir.objConverter.FromUnstructured(config, obj)
+	if err != nil {
+		return fmt.Errorf("failed to convert ClusterRoleBinding object: %w", err)
+	}
+
+	obj.SetNamespace(namespace)
+	obj.Subjects[0].Namespace = namespace
+
+	err = ir.kubeCtl.Create(ctx, obj)
+	if err != nil {
+		return fmt.Errorf("failed to add CluserRoleBinding into cluster: %w", err)
+	}
+
+	return nil
+}
+
+func (ir *InstallResources) addDeployment(ctx context.Context, config map[string]interface{}, namespace string, watch []string) error {
+	obj := &appsv1.Deployment{}
+	err := ir.objConverter.FromUnstructured(config, obj)
+	if err != nil {
+		return fmt.Errorf("failed to convert Deployment object: %w", err)
+	}
+
+	obj.SetNamespace(namespace)
+
+	atlasDomain := os.Getenv("MCLI_OPS_MANAGER_URL")
+	if atlasDomain == "" {
+		atlasDomain = "https://cloud.mongodb.com/"
+	}
+	for i := range obj.Spec.Template.Spec.Containers[0].Args {
```

Comment:
By any chance, could it be that a Deployment won't have any containers, and this code crash with the index out of bounds?